File: src/3402.minimum-operations-to-make-columns-strictly-increasing.3691/solution.java
class Solution {
    public int minimumOperations(int[][] grid) {
        int M = grid.length;
        int N = grid[0].length;
        int count = 0;
        for(int j=0;j<N;j++){
            for(int i=1;i<M;i++){
                if(grid[i][j]<=grid[i-1][j]){
                    int target = grid[i-1][j]+1;
                    count += target-grid[i][j];
                    grid[i][j] = target;
                }
            }
        }
        return count;
    }
}

File: src/3402.minimum-operations-to-make-columns-strictly-increasing.3691/solution.ts
function minimumOperations (grid: number[][]): number {
    let result = 0;
    const M = grid.length;
    const N = grid[0].length;
    for (let j = 0; j < N; j++) {
        for (let i = 1; i < M; i++) {
            if (grid[i][j] <= grid[i - 1][j]) {
                result += grid[i - 1][j] + 1 - grid[i][j];
                grid[i][j] = grid[i - 1][j] + 1;
            }
        }
    }
    return result;
}
